### PR TITLE
tests/usbredir: Disable test for s390x

### PR DESCRIPTION
**Pull request by jschintag**
USB is not supported on s390x.

### What this PR does
Before this PR:
Building s390x fails, as it wants to pull in code that is unsupported on s390x
After this PR:
Building s390x works

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [x] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [x] PR: The PR description is expressive enough and will help future contributors
- [x] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [x] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [x] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [x] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [x] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [x] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: tests/BUILD.bazel [hunk 1]
```diff
@@ -272,5 +271,43 @@ go_test(
...
+            "//pkg/virtctl/usbredir:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:ppc64le": [
+            "//pkg/virtctl/usbredir:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:riscv64": [
+            "//pkg/virtctl/usbredir:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:wasm": [
+            "//pkg/virtctl/usbredir:go_default_library",
```
Comment (brianmcarey) [inline tests/BUILD.bazel:309]: Can we not just add for the arches that are supported by kubevirt?

Comment (jschintag) [reply tests/BUILD.bazel:309]: I just ran `make generate` and it made it this way. I think same thing had been done by `pkg/virtctl/usbredir`

**Review by brianmcarey — COMMENTED** [review]
/lgtm 

thanks @jschintag - we should look at making the s390x build lane required

File: tests/usbredir_test.go [hunk 2]
```diff
@@ -1,3 +1,5 @@
+//go:build !s390x
```
Comment (xpivarc) [inline tests/usbredir_test.go:1]: We use labels in order to filter tests.

Comment (jschintag) [reply tests/usbredir_test.go:1]: The Problem in this case is, that the usbredir code itself is not build on s390x, so the whole function test does not compile on s390x when including this, as it is referencing code that does not exist on s390x.

File: tests/BUILD.bazel (hunk 1 above)
Comment (victortoso) [reply tests/BUILD.bazel:309]: Yeah, this come with s390x result. I was surprised by the verbosity of it as well but I guess with can improve later if needed

https://github.com/kubevirt/kubevirt/commit/49509be6cb238dfcb421aac2b5abf00c60dbf355#diff-f654fc024b6380eba90cf25089c9b77c2ad043a2523a21fac46f7e5aaf031c3bR16

File: tests/usbredir_test.go (hunk 2 above)
Comment (victortoso) [reply tests/usbredir_test.go:1]: Yes, I didn't considered unsupported platforms when writing this.. My expectation is that on s390x the tests are either not run or run and receive `unsupported platform` kind of error somewhere. The latter would need changes in both `virtctl/usbredir` and `tests/usbredir_test.go`... so I'd go with former (this PR) for now.

**Comment by victortoso** [comment]
> Before this PR:
> Building s390x fails, as it wants to pull in code that is unsupported on s390x
> After this PR:
> Building s390x works

I was thinking how we missed this when introducing s390x. We don't have proper CI for it yet, correct?

**Comment by jschintag** [comment]
> > Before this PR:
> > Building s390x fails, as it wants to pull in code that is unsupported on s390x
> > After this PR:
> > Building s390x works
> 
> I was thinking how we missed this when introducing s390x. We don't have proper CI for it yet, correct?

Well, we do have jobs for cross-compiling and unit-tests, but we didn't have the cross compiling working when merging the big PR, so of course no CI was running.
I think your changes merged nearly at the same time as the cross-compile rework.

**Comment by victortoso** [comment]
As per https://github.com/kubevirt/kubevirt/pull/12177#discussion_r1650683273
/lgtm

File: tests/usbredir_test.go (hunk 2 above)
Comment (xpivarc) [reply tests/usbredir_test.go:1]: I see, I forgot we import the `virtctl/usbredir` and that is set to not be build... I do wonder why we want to not build the package, @victortoso ? Imho the more straightforward thing is to not include the command. There is nothing in the package that would disallow it to be build on s390x. Then we could remain consistent here and use labels.

**Comment by xpivarc** [comment]
/approve
Even tho I would prefer to just enable building of the package and use labels.

File: tests/usbredir_test.go (hunk 2 above)
Comment (victortoso) [reply tests/usbredir_test.go:1]: s390x is big-endian and that's, afaict, is not handled by usbredir protocol.
we can still _build_ on s390x but we should error out that platform is unsupported on runtime.
host-size, qemu's usbredir device driver is not included in [c9s packages](https://mirror.stream.centos.org/9-stream/AppStream/s390x/os/Packages/)

Comment (jschintag) [reply tests/usbredir_test.go:1]: Yes, USB is not supported on s390x. The current behavior for kubevirt is to display an error client side in virtctl for usbredir and additionally return an error in the virt-api when trying to use it against an s390x VM.

**Pull request merged**